app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DatasetsOverviewActivity.kt

```diff
@@ -3,29 +3,41 @@ package com.github.HumanLearning2021.HumanLearningApp.view.dataset_editing
 import android.content.Intent
 import android.os.Bundle
 import android.util.Log
-import android.view.View
+import android.widget.Button
 import androidx.appcompat.app.AppCompatActivity
 import com.github.HumanLearning2021.HumanLearningApp.R
+import com.github.HumanLearning2021.HumanLearningApp.view.MainActivity
 import com.github.HumanLearning2021.HumanLearningApp.view.dataset_list_fragment.DatasetListWidget
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class DatasetsOverviewActivity : AppCompatActivity() {
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_datasets_overview)
 
-        val dsListFragment = supportFragmentManager.findFragmentById(R.id.dataOverview_fragment)
+        val dsListFragment = supportFragmentManager.findFragmentById(R.id.datasetsOverview_fragment)
         if (dsListFragment is DatasetListWidget) {
             dsListFragment.selectedDataset.observe(this) {
-                Log.d("DataOverview activity", "selected ds is  $it")
-                // TODO include Dataset (it) with Intent
-                startActivity(Intent(this, CategoriesEditingActivity::class.java))
+                Log.d(this.localClassName, "Selected dataset $it")
+                startActivity(
+                    Intent(this, DisplayDatasetActivity::class.java)
+                        .putExtra("dataset_id", it.id as String)
+                )
             }
         }
+
+        setCreateNewDSButtonListener()
+    }
+
+    private fun setCreateNewDSButtonListener() {
+        findViewById<Button>(R.id.datasetsOverviewButton).setOnClickListener {
+            startActivity(Intent(this, CategoriesEditingActivity::class.java))
+        }
     }
 
-    fun launchDataCreationActivity(@Suppress("UNUSED_PARAMETER") view: View){
-        startActivity(Intent(this, CategoriesEditingActivity::class.java))
+    override fun onBackPressed() {
+        startActivity(Intent(this, MainActivity::class.java))
     }
```

Comment:
I think there's a cleaner way to do this, but it's ok for now.
https://developer.android.com/guide/topics/manifest/activity-element#parent

Reply:
Oh that's way better indeed. I prefer not doing this in this PR because I used a lot of those "onBackPressed" but I will modify this for next sprint. TY!